Py/DICOMSegmentationPlugin.py

```diff
@@ -295,8 +299,276 @@ def load(self,loadable):
 
     # TODO: the outputDir should be cleaned up
 
+    if hasattr(slicer.modules, 'segmentations'):
+
+      import vtkSlicerSegmentationsModuleLogic
+      import vtkSlicerSegmentationsModuleMRML
+      import vtkSegmentationCore
+
+      segmentationNode = vtkSlicerSegmentationsModuleMRML.vtkMRMLSegmentationNode()
+      segmentationNode.SetName(seriesName)
+      segmentationNode.AddNodeReferenceID('colorNodeID', segmentationColorNode.GetID())
+      slicer.mrmlScene.AddNode(segmentationNode)
+
+      segmentationDisplayNode = vtkSlicerSegmentationsModuleMRML.vtkMRMLSegmentationDisplayNode()
+      segmentationNode.SetAndObserveDisplayNodeID(segmentationDisplayNode.GetID())
+      slicer.mrmlScene.AddNode(segmentationDisplayNode)
+
+      segmentation = vtkSegmentationCore.vtkSegmentation()
+      segmentation.SetMasterRepresentationName(vtkSegmentationCore.vtkSegmentationConverter.GetSegmentationBinaryLabelmapRepresentationName())
+
+      segmentationNode.SetAndObserveSegmentation(segmentation)
+      self.addSeriesInSubjectHierarchy(loadable, segmentationNode)
+
+      colorID = 1
+      for segmentNode in segmentNodes:
+        segment = vtkSegmentationCore.vtkSegment()
+        segment.SetName(segmentNode.GetName())
+
+        segmentColor = [0,0,0,0]
+        segmentationColorNode.GetColor(colorID, segmentColor)
+        segment.SetDefaultColor(segmentColor[0:3])
+
+        colorID += 1
+
+        #TODO: when the logic class is created, this will need to be changed
+        logic = vtkSlicerSegmentationsModuleLogic.vtkSlicerSegmentationsModuleLogic()
+        orientedImage = logic.CreateOrientedImageDataFromVolumeNode(segmentNode)
+        segment.AddRepresentation(vtkSegmentationCore.vtkSegmentationConverter.GetSegmentationBinaryLabelmapRepresentationName(), orientedImage)
+        segmentation.AddSegment(segment)
+
+        segmentDisplayNode = segmentNode.GetDisplayNode()
+        slicer.mrmlScene.RemoveNode(segmentDisplayNode)
+        slicer.mrmlScene.RemoveNode(segmentNode)
+
+      segmentation.CreateRepresentation(vtkSegmentationCore.vtkSegmentationConverter.GetSegmentationClosedSurfaceRepresentationName(), True)
+      slicer.mrmlScene.RemoveNode(mergeNode)
+
     return True
 
+  def examineForExport(self, node):

```

Comment:
somewhere here should be a check that the image volume this segmentation corresponds to has DICOM instances, otherwise segmentation would not be exportable into DICOM SEG (at least, not with this plugin).